Tests/MSFT_xClusterDisk.Tests.ps1

```diff
@@ -12,191 +7,248 @@ if (!$PSScriptRoot)
 $RootPath   = (Resolve-Path -Path "$PSScriptRoot\..").Path
 $ModuleName = 'MSFT_xClusterDisk'
 
-Add-WindowsFeature -Name RSAT-Clustering-PowerShell -ErrorAction SilentlyContinue    
+try
```

Comment:
Don't use catch to detect the OS Product Type.  Query it.  Code for WMF5.0 and lower where DSC is supported:
```PowerShell
if((Get-CimInstance 'Win32_OperatingSystem').ProductType -ne 1)
{
    # For server operating system
    Add-WindowsFeature -Name RSAT-Clustering-PowerShell -ErrorAction Stop
}
else
{
    # For client operating system
    Enable-WindowsOptionalFeature -Online -FeatureName 'RSATClient-Features-Clustering' -ErrorAction Stop
}
```
On WMF 5.1, `Get-ComputerInfo` will have more information about nano, but I don't think it is needed here.